main/tasks/ping_task.cpp

```diff
@@ -0,0 +1,195 @@
+#include "ping_task.h"
+#include "esp_log.h"
+#include "ping/ping_sock.h"
+#include "lwip/inet.h"
+#include "lwip/ip_addr.h"
+#include "lwip/sockets.h"
+#include "nvs_config.h"
+#include "global_state.h"
+#include "stratum_task.h"
+
+// Configuration for ping intervals and history
+#define PING_COUNT 7           // number of pings per round
+#define PING_INTERVAL_MS 1000  // delay between individual pings in ms
+#define PING_TIMEOUT_MS 1000   // timeout per ping request in ms
+#define PING_DELAY 60          // delay between ping rounds in seconds
+
+static const char *TAG = "ping task";
+static double last_ping_rtt_ms = 0.0;
+
+// Result structure used by ping logic
+struct PingResult {
+    bool success;
+    uint16_t replies;
+    double avg_rtt_ms;
+    double min_rtt_ms;
+    double max_rtt_ms;
+};
+
+// Run a ping session using resolved IP from STRATUM_MANAGER
+PingResult perform_ping(const char* ip_str, const char* hostname_str) {
+    PingResult result = { .success = false, .replies = 0, .avg_rtt_ms = 0.0, .min_rtt_ms = 1e6, .max_rtt_ms = 0.0 };
+
+    ip_addr_t target_addr;
+    if (!ip_str || !ipaddr_aton(ip_str, &target_addr)) {
+        ESP_LOGE(TAG, "Invalid IP string passed to perform_ping (%s)", ip_str ? ip_str : "null");
+        return result;
+    }
+
+    struct PingStats {
+        uint16_t replies;
+        uint32_t total_time_ms;
+        double min_rtt;
+        double max_rtt;
+        const char* hostname;
+        bool header_shown;
+    } stats = { .min_rtt = 1e6, .max_rtt = 0.0, .hostname = hostname_str, .header_shown = false };
+
+    // Configure ping session
+    esp_ping_config_t config = ESP_PING_DEFAULT_CONFIG();
+    config.count = PING_COUNT;
+    config.interval_ms = PING_INTERVAL_MS;
+    config.timeout_ms = PING_TIMEOUT_MS;
+    config.target_addr = target_addr;
+
+    // Set callback to accumulate successful replies
+    static esp_ping_callbacks_t cbs = {};
```

Comment:
one thing, could you please explain why you made it `static`? I have a little bit of a bad feeling and normally I only use static if it really needs to be static :thinking: